serde_tests/tests/token.rs

```diff
@@ -429,6 +429,10 @@ impl de::Error for Error {
 
     fn end_of_stream() -> Error { Error::EndOfStreamError }
 
+    fn invalid_value(msg: &str) -> Error {
+        Error::InvalidValue(msg.to_owned())
```

Comment:
I think `de::Error` deserves a new method `duplicate_field`. We already have `unknown_field`. The default impl could forward to `invalid_value` for now, and we can add an issue for 0.8 to remove that default impl.